hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/engine/StepRunner.java

```diff
@@ -99,10 +99,24 @@ public static boolean isStepRunnerCurrentlyExecutingOnPartitionThread() {
         return CURRENTLY_EXECUTING_ON_PARTITION_THREAD.get();
     }
 
-    private void addOpToOffloadedOps(MapOperation op) {
-        op.setOperationResponseHandler(new OffloadedStepResponseHandler(op.getOperationResponseHandler()));
-        offloadedOperations.add(op);
-        op.getRecordStore().incMapStoreOffloadedOperationsCount();
+    private boolean addOpToOffloadedOps(MapOperation op) {
+        if (offloadedOperations.add(op)) {
+            op.getRecordStore().incMapStoreOffloadedOperationsCount();
+            return true;
+        }
+
+        return false;
+    }
+
+    private void setSteppedOpResponseHandler() {
```

Comment:
Previously it was just before `offloadedOperations.add(op)`, now it's in the `run()`. Why this change is made? I am concerned about possibility of failure (or sending response) before this method is called and operations left in the `offloadedOperations` set infinitely.

Reply:
It will be a bug if something happen like that. Only StepRunner can execute and remove from offloadedOperations. Retries was breaking this and this PR is a fix that. Tried to explain this in PR description.